### PR TITLE
refactor: `SignOrExecute` elements

### PR DESCRIPTION
**Pull request by katspaugh**
## What it solves

Part of #1828

An umbrella PR for code refactors in preparation for a tx flow redesign.

## How to test
Make all kinds of transactions, immediate executions/signatures/on-chain signatures/relayed executions. Everything should work like before.

## Analytics changes
Relayed executions now have their own event instead of "Propose".

### CODE REVIEW

File: src/components/library/index.tsx
Comment (usame-algan) [inline src/components/library/index.tsx:1]: Is this something we want to use more in the future e.g. for isolated testing of new components?

**Review by usame-algan — APPROVED** [review]

File: src/services/analytics/events/wallet.ts [hunk 1]
```diff
@@ -35,4 +35,9 @@ export const WALLET_EVENTS = {
     action: 'Confirm message',
     category: WALLET_CATEGORY,
   },
+  RELAYED_EXECUTION: {
+    event: EventType.META,
+    action: 'Relayed execution',
+    category: WALLET_CATEGORY,
+  },
```
Comment (usame-algan) [inline src/services/analytics/events/wallet.ts:42]: Technically the `TxEvent.RELAYING` event is dispatched before it was successfully executed. Should we rename this to `Relaying execution` accordingly?

Comment (katspaugh) [reply src/services/analytics/events/wallet.ts:42]: It's not a verb in past tense, it's supposed to mean "An execution that's being relayed"

File: src/components/library/index.tsx
Comment (katspaugh) [reply src/components/library/index.tsx:1]: Oops, I didn't mean to merge that.

**Review by iamacook — APPROVED** [review]

**Comment by liliya-soroka** [comment]
I am getting this error from time to time on sign+execute or sign without execution. When the submit is clicked the second time, the signing or execution passes without any problem
image

**Comment by liliya-soroka** [comment]
The app crash when I try to execute - https://tx_refactor--webcore.review-web-core.5afe.dev/transactions/tx?safe=gno:0x9DF58F04F3e914ED8D972146401C85313A53FecA&id=multisig_0x9DF58F04F3e914ED8D972146401C85313A53FecA_0xf46cbeeb673201593a8ffae2caa7ffb7877fc3671599c96f6135993271e924b9

https://user-images.githubusercontent.com/338622/233076734-34ed9561-c114-4e3b-a148-a91af276548f.mov

**Comment by usame-algan** [comment]
>  The app crashes when I try to execute - https://tx_refactor--webcore.review-web-core.5afe.dev/transactions/tx?safe=gno:0x9DF58F04F3e914ED8D972146401C85313A53FecA&id=multisig_0x9DF58F04F3e914ED8D972146401C85313A53FecA_0xf46cbeeb673201593a8ffae2caa7ffb7877fc3671599c96f6135993271e924b9

Looks like `data` can be null which is not reflected in our types.

@liliya-soroka can you provide the steps in how you created that transaction?

**Comment by liliya-soroka** [comment]
@usame-algan , added to the comment with the issue

**Comment by usame-algan** [comment]
@liliya-soroka I was able to reproduce the bug on dev as well. I suggest to fix it there first. Will create an issue for it.

**Comment by liliya-soroka** [comment]
What was tested:
1. Execution using relay on Gnosis chain
2. Execution when relay tx is 0 of 5 -> execution with the owner 
3. outgoing transfer /transferfrom
4. spending limits /module txs
5. settings txs
6. safe apps txs with diff amounts of actions
7. sign + execute in N of M
8. execute in N of M
9. sign + execute in 1 of M
10.  execute in 1 of M
11. batch tx execution  without relay 
12. batch tx  execution with relay - we have separate issue not related to this PR

**Pull request merged**